### PR TITLE
[WebGPU] support UnsortedSegmentSum kernel

### PR DESCRIPTION
**Pull request by xhcao**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

- - -
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/7495)

### CODE REVIEW

File: tfjs-backend-webgpu/src/kernels/UnsortedSegmentSum.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  const inSize = util.sizeFromShape([permutedX.shape[axis]]);
+  const a2D =
+      reshape({inputs: {x: permutedX}, backend, attrs: {shape: [-1, inSize]}});
+  toDispose.push(a2D);
+
+  const dtype = x.dtype;
+  const output = fill(
+      {backend, attrs: {shape: [a2D.shape[0], numSegments], value: 0, dtype}});
+  const program = new UnsortedSegmentSumProgram(
+      [a2D.shape[0], a2D.shape[1]], numSegments, dtype);
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/UnsortedSegmentSum.ts:60]: `[a2D.shape[0], a2D.shape[1]]` -> `a2D.shape` ?

File: tfjs-backend-webgpu/src/unsorted_segment_sum_webgpu.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  dispatch: [number, number, number];
+  variableNames = ['x', 'segmentIds'];
+  uniforms = 'numSegments : i32, xSize: i32,';
+  workgroupSize: [number, number, number] = [64, 1, 1];
+  size = true;
+  atomic = true;
+  type: DataType;
+
+  constructor(
+      inSize: [number, number], numSegments: number, outputDtype: DataType) {
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/unsorted_segment_sum_webgpu.ts:37]: `inSize` -> `inShape`? we usually use size as a number. eg. sizeFromShape

File: tfjs-backend-webgpu/src/unsorted_segment_sum_webgpu.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+export class UnsortedSegmentSumProgram implements WebGPUProgram {
+  outputShape: number[] = [];
+  shaderKey: string;
+  dispatchLayout: {x: number[]};
+  dispatch: [number, number, number];
+  variableNames = ['x', 'segmentIds'];
+  uniforms = 'numSegments : i32, xSize: i32,';
+  workgroupSize: [number, number, number] = [64, 1, 1];
+  size = true;
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/unsorted_segment_sum_webgpu.ts:32]: It seems that `size = true` can be removed.

**Review by qjia7 — APPROVED** [review]
LGTM, thanks!

**Review by gyagp — APPROVED** [review]
LGTM

**Pull request merged**